### ISSUE TITLE
Front end build step abstraction

### ISSUE THREAD

**Issue by kurtisassad** [hicommonwealth/commonwealth#9275]
Now that we bundle the front end inside github CI, we have the opportunity to go for promotion based deployments. The issue right now is that we hardcode the front end environment variables into the docker image itself. But this is coupled with a specific environment (Frick/frack/...). 

In order to set up the promotion based deployments, we need to come up with a way to inject the environment variables into the front end bundle at deployment time.

**Comment by rbennettcw** [comment]
Yeah it's an issue because the env vars get baked in the JS bundle at build time.

We'll need to inject the docker container env vars into the app at container runtime. That means modifying/augmenting code at runtime.
- We probably don't want to modify the JS bundle at runtime because that's pretty hacky/brittle
- But maybe we could update the `index.html` file at runtime to read env vars and add them as global variables as described here: https://www.freecodecamp.org/news/how-to-implement-runtime-environment-variables-with-create-react-app-docker-and-nginx-7f9d42a91d70/
- Since the `index.html` file is served by the backend, we could probably update the backend to handle the injection:
```js
  app.get('*', (req: Request, res: Response) => {
    // get env vars, get html string, inject envs into html via substitution
    res.sendFile(`${__dirname}/index.html`);
  });
```
Feels a little odd since it's blending BE/FE responsibilities, but we're already doing that anyway.

**Comment by timolegros** [comment]
> Yeah it's an issue because the env vars get baked in the JS bundle at build time.
> 
> We'll need to inject the docker container env vars into the app at container runtime. That means modifying/augmenting code at runtime.
> 
> * We probably don't want to modify the JS bundle at runtime because that's pretty hacky/brittle
> * But maybe we could update the `index.html` file at runtime to read env vars and add them as global variables as described here: [freecodecamp.org/news/how-to-implement-runtime-environment-variables-with-create-react-app-docker-and-nginx-7f9d42a91d70](https://www.freecodecamp.org/news/how-to-implement-runtime-environment-variables-with-create-react-app-docker-and-nginx-7f9d42a91d70/)
> * Since the `index.html` file is served by the backend, we could probably update the backend to handle the injection:
> 
> ```js
>   app.get('*', (req: Request, res: Response) => {
>     // get env vars, get html string, inject envs into html via substitution
>     res.sendFile(`${__dirname}/index.html`);
>   });
> ```
> 
> Feels a little odd since it's blending BE/FE responsibilities, but we're already doing that anyway.

This approach would block us from serving the bundle from S3 no?
